diff --git a/src/piston_engine_simulator.cpp b/src/piston_engine_simulator.cpp
index c0a2e3e..2219d56 100644
--- a/src/piston_engine_simulator.cpp
+++ b/src/piston_engine_simulator.cpp
@@ -33,7 +33,6 @@ PistonEngineSimulator::~PistonEngineSimulator() {
     assert(m_crankshaftFrictionConstraints == nullptr);
     assert(m_exhaustFlowStagingBuffer == nullptr);
     assert(m_delayFilters == nullptr);
-    assert(m_antialiasingFilters == nullptr);
 }
 
 void PistonEngineSimulator::loadSimulation(Engine *engine, Vehicle *vehicle, Transmission *transmission) {
