diff --git a/include/constraint.h b/include/constraint.h
index 91dd385..dc15852 100644
--- a/include/constraint.h
+++ b/include/constraint.h
@@ -28,7 +28,7 @@ namespace atg_scs {
             virtual ~Constraint();
 
             virtual void calculate(Output *output, SystemState *state);
-            __forceinline int getConstraintCount() const { return m_constraintCount; }
+            inline int getConstraintCount() const { return m_constraintCount; }
 
             int m_index;
             int m_bodyCount;
diff --git a/include/gaussian_elimination_sle_solver.h b/include/gaussian_elimination_sle_solver.h
index ba5e697..e3404aa 100644
--- a/include/gaussian_elimination_sle_solver.h
+++ b/include/gaussian_elimination_sle_solver.h
@@ -16,7 +16,7 @@ namespace atg_scs {
                     Matrix *result,
                     Matrix *previous);
 
-            static __forceinline double fastAbs(double v) {
+            static inline double fastAbs(double v) {
                 return (v > 0)
                     ? v
                     : -v;
diff --git a/include/matrix.h b/include/matrix.h
index 591176f..f8b1e28 100644
--- a/include/matrix.h
+++ b/include/matrix.h
@@ -17,21 +17,21 @@ namespace atg_scs {
 
             void set(const double *data);
 
-            __forceinline void set(int column, int row, double value) {
+            inline void set(int column, int row, double value) {
                 assert(column >= 0 && column < m_width);
                 assert(row >= 0 && row < m_height);
 
                 m_matrix[row][column] = value;
             }
 
-            __forceinline void add(int column, int row, double value) {
+            inline void add(int column, int row, double value) {
                 assert(column >= 0 && column < m_width);
                 assert(row >= 0 && row < m_height);
 
                 m_matrix[row][column] += value;
             }
 
-            __forceinline double get(int column, int row) {
+            inline double get(int column, int row) {
                 assert(column >= 0 && column < m_width);
                 assert(row >= 0 && row < m_height);
 
@@ -60,7 +60,7 @@ namespace atg_scs {
             int getWidth() const { return m_width; }
             int getHeight() const { return m_height; }
 
-            __forceinline void fastRowSwap(int a, int b) {
+            inline void fastRowSwap(int a, int b) {
                 double *temp = m_matrix[a];
                 m_matrix[a] = m_matrix[b];
                 m_matrix[b] = temp;
diff --git a/include/sparse_matrix.h b/include/sparse_matrix.h
index c8875c6..b417b8c 100644
--- a/include/sparse_matrix.h
+++ b/include/sparse_matrix.h
@@ -1,12 +1,13 @@
 #ifndef ATG_SIMPLE_2D_CONSTRAINT_SOLVER_SPARSE_MATRIX_H
 #define ATG_SIMPLE_2D_CONSTRAINT_SOLVER_SPARSE_MATRIX_H
 
+#include "matrix.h"
+
 #include <assert.h>
 #include <stdint.h>
 #include <string.h>
 
 namespace atg_scs {
-    class Matrix;
 
     template <int T_Stride = 3, int T_Entries = 2>
     class SparseMatrix {
@@ -261,8 +262,8 @@ namespace atg_scs {
                 }
             }
 
-            __forceinline int getWidth() const { return m_width; }
-            __forceinline int getHeight() const { return m_height; }
+            inline int getWidth() const { return m_width; }
+            inline int getHeight() const { return m_height; }
 
         protected:
             double **m_matrix;
